serve/mlc_serve/model/tvm_model.py

```diff
@@ -446,7 +446,7 @@ def init_tvm_model(
         LOG.info("Running memory profiling.")
         num_blocks = get_num_cache_blocks(
             model,
-            [engine_config.max_input_len] * engine_config.max_num_sequences,
+            [1] * engine_config.max_num_batched_tokens,
```

Comment:
This change will make memory profiling overestimate memory usage and allocate smaller KV cache. I tested with mistral 7B on single H100, the number of blocks reduces from 28638 to 28389

Reply:
I remember doing this change and wondering why the profile result becomes different. Do you know why?

Reply:
There will be more rows in the output logits (https://github.com/octoml/mlc-llm/blob/66a2e534ab4f6981acb0be117b99ca79ea862578/mlc_llm/relax_model/llama_batched_vllm.py#L481-L496). This seems to be the only place affected by the `seq_lens`

Reply:
in the speculative decoding or logits workflow, it can be important and might lead to OOM issue

Reply:
> in the speculative decoding or logits workflow, it can be important and might lead to OOM issue

@elvin-n Can you elaborate how OOM could occur? I believe this change makes memory profiling over-estimate memory consumption, thus allocating less KV cache blocks. In the worst case, the profiling is so conservative that it allocates not enough blocks for the max_context_length or even OOM during profiling. But it shouldn't OOM after server startup.